### ISSUE TITLE
i18next.on("missingKey", () => {})

### ISSUE THREAD

**Issue by jagretz** [i18next/i18next#1082]
The [missingKey API](https://www.i18next.com/overview/api#onmissingkey) seems wrong, but perhaps I am using the API incorrectly.

In a nutshell I don't think the [missingKey](https://www.i18next.com/overview/api#onmissingkey) event handler is working properly. Here is an example:

Mind you, I am using this with react-i18next and pulling in `reactI18nextModule` but this seems to be an issue with the i18next API and not react-specifics.

Essentially, when I add an event handler for "missingKey" it is not triggered when I supply a missing key. My expectation, of course, is that the console would log `"event > missingKey"` whenever it encounters a missing key.

Example code is provided for reference:
```js
import i18next from "i18next";

i18next
    .use(reactI18nextModule)
    .init(/* options */)
    .on('missingKey', () => console.log("event > missingKey"))
	
/* ...somewhere in the react tree... t() is invoked */

// key "unknown" does not exist for any ns, or language 
t("unknown")
```

Instead of logging, nothing happens; the event handler is never invoked. However, if in the options I enable both `saveMissing` and `missingKeyHandler` the `.on('missingKey'...)` will trigger -- that certainly isn't intuitive and according to the docs would be inappropriate for the intended use, see [missing keys](https://www.i18next.com/overview/configuration-options#missing-keys).

Appropriate or not, setting `saveMissing` and `missingKeyHandler` does trigger the `missingKey` event. 

_unrelated..._
I have a lot of components rendering the same key. If it is missing once, then it is always missing until updated with a value. I don't need to see the message more than one time. Any way around this without some custom hacking and memoization?

**Comment by jamuhl** [comment]
a) missingKeyHandler description even states that you need to enable saveMissing - might be not intuitive but it's the way it is -> basically just an alternative to use a backend supporting that (xhr, locize,... do this -> and forward the missing to backend automatically without custom handler)

b) the backend would handle that automatically - but you will have to do https://github.com/i18next/i18next/blob/master/src/BackendConnector.js#L199 yourself

**Comment by jagretz** [comment]
So the idea is that `on('missingKey'...)` is only invoked _if_ you've set the `saveMissing` and `missingKeyHandler` in the configuration?

We only use a backend service to retrieve translations, not to provide an fallback value or to update a translations.

**Comment by jamuhl** [comment]
yes...https://github.com/i18next/i18next/blob/master/src/Translator.js#L148

will update docs

**Comment by jagretz** [comment]
Gotcha, thanks for the clarification @jamuhl !

**Comment by jamuhl** [comment]
if you need some different behaviour we could check if we could add that...think wouldn't be to much trouble to change that behaviour
